src/components/FileUpload.tsx

```diff
@@ -1,41 +1,58 @@
-import React, { useState } from 'react';
+import React, { useCallback, useState } from 'react';
 import { Input } from '@/components/ui/input';
 import { Label } from '@/components/ui/label';
 import Preview from '@/components/Preview';
 import { useToast } from '@/components/ui/use-toast';
+import axios from 'axios';
 
 const FileUpload = () => {
   const { toast } = useToast();
   const inputRef = React.useRef<HTMLInputElement>(null);
-  const [pdf, setPdf] = useState<string | null>(null);
+  const [filePath, setFilePath] = useState<string | null>(null);
+  const [pdf, setPdf] = useState<string | Blob>('');
```

Comment:
The type here should only be `Blob` since I don't see a case where this state will ever be a `string`

Reply:
Can't initialize the pdf without having string since null or undefined cannot be assigned to type Blob